kornia/geometry/bbox_v2.py

```diff
@@ -0,0 +1,585 @@
+
+    if points.shape[0] != M.shape[0]:
+        raise ValueError(
+            f"Batch size mismatch. Got {points.shape[0]} for hexahedron, {M.shape[0]} for the transformation matrix."
+        )
+
+    transformed_boxes: torch.Tensor = kornia.transform_points(M, points)
+    transformed_boxes = transformed_boxes.view_as(boxes)
+    return transformed_boxes
+
+
+def _boxes_to_polygons(
+    xmin: torch.Tensor, ymin: torch.Tensor, width: torch.Tensor, height: torch.Tensor
+) -> torch.Tensor:
+    if not xmin.ndim == ymin.ndim == width.ndim == height.ndim == 2:
+        raise ValueError("We expect to create a batch of hexahedron in vertices format (B, N, 4, 2)")
+
+    # Create (B,N,4,2) with all points in top left position of the bounding box
+    polygons = torch.zeros((xmin.shape[0], xmin.shape[1], 4, 2), device=xmin.device, dtype=xmin.dtype)
+    polygons[..., 0] = xmin.unsqueeze(-1)
+    polygons[..., 1] = ymin.unsqueeze(-1)
+    # Shift top-right, bottom-right, bottom-left points to the right coordinates
+    polygons[..., 1, 0] += width  # Top right
+    polygons[..., 2, 0] += width  # Bottom right
+    polygons[..., 2, 1] += height  # Bottom right
+    polygons[..., 3, 1] += height  # Bottom left
+    return polygons
+
+
+def _boxes3d_to_polygons3d(
+    xmin: torch.Tensor,
+    ymin: torch.Tensor,
+    zmin: torch.Tensor,
+    width: torch.Tensor,
+    height: torch.Tensor,
+    depth: torch.Tensor,
+):
+    if not xmin.ndim == ymin.ndim == zmin.ndim == width.ndim == height.ndim == depth.ndim == 2:
+        raise ValueError("We expect to create a batch of 3D hexahedrons in vertices format (B, N, 8, 3)")
+
+    # Front
+    # Create (B,N,4,3) with all points in front top left position of the bounding box
+    front_vertices = torch.zeros((xmin.shape[0], xmin.shape[1], 4, 3), device=xmin.device, dtype=xmin.dtype)
+    front_vertices[..., 0] = xmin.unsqueeze(-1)
+    front_vertices[..., 1] = ymin.unsqueeze(-1)
+    front_vertices[..., 2] = zmin.unsqueeze(-1)
+    # Shift front-top-right, front-bottom-right, front-bottom-left points to the right coordinates
+    front_vertices[..., 1, 0] += width  # Top right
+    front_vertices[..., 2, 0] += width  # Bottom right
+    front_vertices[..., 2, 1] += height  # Bottom right
+    front_vertices[..., 3, 1] += height  # Bottom left
+
+    # Back
+    back_vertices = front_vertices.clone()
+    back_vertices[..., 2] += depth.unsqueeze(-1)
+
+    polygons3d = torch.cat([front_vertices, back_vertices], dim=-2)
+    return polygons3d
+
+
+@torch.jit.script
+class Boxes:
+    def __init__(self, quadrilaterals: torch.Tensor, raise_if_not_floating_point: bool = True):
+        if not quadrilaterals.is_floating_point():
+            if raise_if_not_floating_point:
+                raise ValueError(f"Coordinates must be in floating point. Got {quadrilaterals.dtype}")
+            else:
+                quadrilaterals = quadrilaterals.float()
+
+        if quadrilaterals.numel() == 0:
```

Comment:
```suggestion
        if len(quadrilaterals.shape) == 0:
```
in some old pytorch versions I remember this didn't work